warp10/src/test/java/io/warp10/continuum/egress/EgressFetchHandlerTest.java

```diff
@@ -1,45 +1,74 @@
 package io.warp10.continuum.egress;
 
+import io.warp10.WarpConfig;
 import io.warp10.continuum.gts.GTSDecoder;
 import io.warp10.continuum.gts.GTSEncoder;
 import io.warp10.continuum.store.thrift.data.Metadata;
-import org.apache.commons.io.IOUtils;
-import org.junit.Test;
 
-import java.io.*;
+import java.io.ByteArrayOutputStream;
+import java.io.PrintWriter;
+import java.io.StringReader;
 import java.nio.charset.StandardCharsets;
-import java.util.*;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
 import java.util.concurrent.atomic.AtomicLong;
 import java.util.concurrent.atomic.AtomicReference;
 
+import org.apache.commons.io.IOUtils;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
 import static java.util.Collections.singleton;
 import static java.util.Collections.singletonMap;
 import static org.junit.Assert.assertEquals;
 
 public class EgressFetchHandlerTest {
-  private static ClassLoader CL = EgressFetchHandlerTest.class.getClassLoader();
+
+  private static final ClassLoader CL = EgressFetchHandlerTest.class.getClassLoader();
+
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+    WarpConfig.safeSetProperties(new StringReader("warp.timeunits=us"));
+  }
 
   @Test
-  public void testJsonDump() throws Exception {
+  public void testJsonDumpWithoutExpose() throws Exception {
+    String json = jsonDumpWithExpose(false);
+    String expect = IOUtils.toString(CL.getResourceAsStream("EgressFetchHandlerTest_1.json"), StandardCharsets.UTF_8);
+
+    assertEquals(expect.trim(), json);
+  }
+
+  @Test
+  public void testJsonDumpWithExpose() throws Exception {
+    String json = jsonDumpWithExpose(true);
+    String expect = IOUtils.toString(CL.getResourceAsStream("EgressFetchHandlerExposeTest.json"), StandardCharsets.UTF_8);
+
+    assertEquals(expect.trim(), json);
+  }
+
+  private String jsonDumpWithExpose(boolean expose) throws Exception {
+    Map<String, String> labels = new HashMap<String, String>();
+    labels.put("foo", "bar");
```

Comment:
Would be clearer if the values used for producer and owner were actual UUIDs, it would not lead people into thinking other values could be there

Reply:
done